### ISSUE TITLE
perf_counter_ns() on Windows has changed scale and base reference

### ISSUE THREAD

**Issue by atari2600tim** [fluffos/fluffos#973]
When I updated FluffOS on Windows 10 from one built in early 2021 to current, the `perf_counter_ns()` function changed drastically.  It now counts microseconds instead of nanoseconds.  Also it isn't microseconds since epoch but rather compared to something else.  I'm not sure what it's compared to, it's not very large and sometimes is negative.

https://github.com/fluffos/fluffos/commit/beb357a7eb6fc371d464af0da40510cd724d7953 added an `#ifdef _WIN32` to run some alternate code.  I added three more zeroes there and that got it to run at nanosecond scale again.  But I didn't know quite what to do to make it counted from epoch again.  I found some [documentation](https://learn.microsoft.com/en-us/windows/win32/sysinfo/acquiring-high-resolution-time-stamps) that says **QueryPerformanceCounter** is independent of any external time reference, and to use **GetSystemTimePreciseAsFileTime** for high-resolution time of day measurements.  But the same documentation says that QueryPerformanceCounter was introduced with Windows 7 while GetSystemTimePreciseAsFileTime was introduced with Windows 8, so I would not want to suggest leaving Windows 7 behind just to restore the time of day functionality.

The builds prior to that commit all worked good on my Windows 10 PCs.  I don't know enough about it to say if it would make sense to change the function from QPC to GetSystemTimePreciseAsFileTime, or to just have it use the old code.  If it stays how it is then it should at least be adjusted by some digits to be nano instead of micro.

Anyway, long story short, I have written a change to use the other function, but I hesitate to make a pull request considering it would drop support for Windows 7.  https://github.com/atari2600tim/fluffos/commit/326bba29df6f96012ca401500a78bdd05a703c20 has that.  It might be better to just revert the other change.  But I personally do not understand the advantages that QPC was intended to make compared to the original code, so I do not know which one to suggest.

**Comment by thefallentree** [comment]
The return of this value is only useful  when you take 2 values and
subtract them, it is not intended to be used as a timestamp. see
https://docs.python.org/3/library/time.html#time.perf_counter

What value did you want to get instead?  Do you want nanosecond wall clock
time?

**Comment by atari2600tim** [comment]
I wasn't familiar with the Python function, but you're right and the time since epoch is only added by FluffOS (https://www.fluffos.info/efun/system/perf_counter_ns.html) and not in the Python version that I guess it was inspired by.

I'd prefer the Windows function be same as Linux, which is nanoseconds since midnight (GMT) January 1, 1970.  My Windows build from early 2021 before https://github.com/fluffos/fluffos/commit/beb357a7eb6fc371d464af0da40510cd724d7953 was doing that.

I used it mainly to compare between different calls of the function as expected, so just adjusting micro to nano would fix the scale issue.  But also I have a [ULID](https://github.com/ulid/spec) simul_efun, so I'd prefer keep the clock functionality if possible.  I think just reverting that change from before would make it act like Linux again, but I am not sure if there is some other advantage that I don't understand.

**Comment by thefallentree** [comment]
The document is a bit misleading here perf_counter_ns() is only
intended to be used for measuring length of time,   it's not supposed
to return wall time. it just so happens on linux it's the same clock.

The reason this is changed on Windows is because the original
implementation actually won't return a nanosecond precision clock.

Do you really need a nanosecond level wall clock efun ? If so we need
to add one specifically for that.

**Comment by atari2600tim** [comment]
The old behavior for me on Windows 10 before that ifdef was that it too returned nanosecond wall clock.  Although I don't know how much actual precision it had.  It sounds like QueryPerformanceCounter is indeed better.  Although with .`ns` being in the name, I would have `perf_counter_ns` return nanoseconds instead of microseconds scale.

A new efun specifically for high precision wall clock sounds good, could be named something like `time_ns()`.  On linux it could be same as perf_counter_ns while on Windows it would use GetSystemTimePreciseAsFileTime.
